provisioner/provisioner.go

```diff
@@ -528,7 +524,7 @@ func (p *Provisioner) executeCnspec(ui packer.Ui, comm packer.Communicator) erro
 	var err error
 	if p.config.Incognito {
 		ui.Message("scan packer build in incognito mode")
-		scanService := scan.NewLocalScanner()
+		scanService := scan.NewLocalScanner(scan.WithRecording(providers.NullRecording{}))
```

Comment:
Why do we need to set those? By default this should not record

Reply:
Before this, I got a panic because of a nil reference.

Reply:
@chris-rock The discussed follow-up https://github.com/mondoohq/packer-plugin-cnspec/pull/143